src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java

```diff
@@ -0,0 +1,107 @@
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.nio.file.DirectoryStream.Filter;
+import java.time.LocalDate;
+import java.util.Optional;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Accesses Datastore to support managing ScheduledInterview entities. */
+public class DatastoreScheduledInterviewDao implements ScheduledInterviewDao {
+  // @param datastore the DatastoreService we're using to interact with Datastore.
+  private DatastoreService datastore;
+
+  /** Initializes the fields for ScheduledInterviewDatastoreDAO. */
+  public DatastoreScheduledInterviewDao() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  /**
+   * Retrieve a scheduledInterviewEntity from Datastore
+   * and return it as a scheduledInterview object.
+   */
+  public Optional<ScheduledInterview> get(long id) {
+    Key key = KeyFactory.createKey("id", id);
+    Entity scheduledInterviewEntity; 
+    try {
+      scheduledInterviewEntity = datastore.get(key); 
+    } catch (Exception e) {
+      return Optional.empty(); 
+    }
+  
+    return Optional.of(entityToScheduledInterview(scheduledInterviewEntity));
+  }
+
+  /**
+   * Retrieve all scheduledInterview entities from Datastore that involve a particular user
+   * and return them as a list of scheduledInterview objects.
+   */
+  public List<ScheduledInterview> getForPerson(String email) {
+    FilterPredicate interviewerFilter = new FilterPredicate("interviewer", FilterOperator.EQUAL, email);
+    FilterPredicate intervieweeFilter = new FilterPredicate("interviewee", FilterOperator.EQUAL, email);
+    CompositeFilter compositeFilter = CompositeFilterOperator.and(interviewerFilter, intervieweeFilter);
+    Query query = new Query().setFilter(compositeFilter);
+    PreparedQuery results = datastore.prepare(query); 
+    List<ScheduledInterview> relevantInterviews = new ArrayList<>(); 
+
+    for (Entity entity : results.asIterable()) {
+      relevantInterviews.add(
+        entityToScheduledInterview(entity));   
+    }
+    return relevantInterviews;
+  }
+
+
+  /**
+   * Updates an entity in datastore.
+   */
+  public void update(ScheduledInterview scheduledInterview) {
+    datastore.put(create(scheduledInterview));
+  }
+
+  /**
+   * Creates a ScheduledInterview Entity.
+   */
```

Comment:
Be consistent with these one line javadocs (either /** text  */ OR
/**
text
*/)